File: marketplace-frontend/src/app/main/content/_service/stored-chart.service.ts
import { HttpClient } from '@angular/common/http';
import { Marketplace } from '../_model/marketplace';
import { StoredChart } from '../_model/stored-chart';
import { isNullOrUndefined } from 'util';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class StoredChartService {

  constructor(private http: HttpClient) {
  }

  findById(marketplace: Marketplace, id: string) {
    return this.http.get(`${marketplace.url}/chart/id/${id}`);
  }

  findByTitle(marketplace: Marketplace, title: string) {
    return this.http.get(`${marketplace.url}/chart/title/${title}`);
  }

  findAll(marketplace: Marketplace) {
    return this.http.get(`${marketplace.url}/chart`);
  }

  save(marketplace: Marketplace, storedChart: StoredChart) {
    return this.http.post(`${marketplace.url}/chart`, storedChart);
  }
}

File: marketplace/marketplace-core/src/main/java/at/jku/cis/iVolunteer/marketplace/diagram/StoredChartService.java
package at.jku.cis.iVolunteer.marketplace.diagram;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import at.jku.cis.iVolunteer.model.diagram.StoredChart;

@Service
public class StoredChartService {

	@Autowired
	private StoredChartRepository storedChartRepository;

	public void createChart(StoredChart storedChart) {
		StoredChart chart = storedChartRepository.findByTitle(storedChart.getTitle());
		if (chart != null) {
			chart.setData(storedChart.getData());
			storedChartRepository.save(chart);
		} else {
			storedChartRepository.save(storedChart);
		}
	}

	public StoredChart getChart(String id) {
		return storedChartRepository.findOne(id);
	}

	public StoredChart getChartByTitle(String title) {
		return storedChartRepository.findByTitle(title);
	}

	public List<StoredChart> getCharts() {
		List<StoredChart> findAll = storedChartRepository.findAll();
		return findAll;
	}

}
